mitosheet/mitosheet/api/get_imported_files_and_dataframes.py

```diff
@@ -0,0 +1,108 @@
+    parameters to steps that only create a single dataframe.
+
+    This is useful because the frontend allows users to update dataframe imports one 
+    be one - and so having each step only do a single import is convenient for working
+    with the data on the frontend.
+    """
+    if step_type == SimpleImportStepPerformer.step_type():
+        for index, file_name in enumerate(params['file_names']):
+            return [{
+                'step_type': step_type,
+                'params': {
+                    'file_names': [file_name],
+                    'delimeters': get_sublist_at_index_from_optional_list(params.get('delimeters', None), index),
+                    'encodings': get_sublist_at_index_from_optional_list(params.get('encodings', None), index),
+                    'error_bad_lines': get_sublist_at_index_from_optional_list(params.get('error_bad_lines', None), index)
+                }
+            }]
+
+    if step_type == ExcelImportStepPerformer.step_type():
+        for index, sheet_name in enumerate(params['sheet_names']):
+            return [{
+                'step_type': step_type,
+                'params': {
+                    'file_name': params['file_name'],
+                    'sheet_names': [sheet_name],
+                    'has_headers': params['has_headers'],
+                    'skiprows': params['skiprows']
+                }
+            }]
+
+    if step_type == DataframeImportStepPerformer.step_type():
+        for index, df_name in enumerate(params['df_names']):
+            return [{
+                'step_type': step_type,
+                'params': {
+                    'df_names': [df_name],
+                }
+            }]
+
+    return []
+
+
+def get_imported_files_and_dataframes(params: Dict[str, Any], steps_manager: StepsManagerType) -> str:
+    """
+    Returns a list all the imported files and dataframes, and their import params. Does so by turning
+    the import into a list of imports that only import a single dataframe, as this is convenient
+    for what we work with on the frontend.
+
+    [
+        {
+            step_id: string,
+            imports: ({
+                step_type: 'simple_import'
+                params: CSVImportParams
+            } |
+            {
+                step_type: 'excel_import'
+                params: ExcelImportParams
+            } |
+            {
+                step_type: 'dataframe_import'
+                params: DataframeImportParams
+            })[]
+        }
+	]
+	
+    """
+
+    # Then, we turn import steps into a version of the step that just creates a single dataframe
+    # as this is what is easiest to work with on the frontend
+    import_steps_with_just_one_dataframe = []
+    for step in steps_manager.steps_including_skipped:
+        if is_import_step_type(step.step_type):
+            import_steps_with_just_one_dataframe.append({
+                'step_id': step.step_id,
```

Comment:
I don't understand why we use the actual step_id here and we use `fake_id` in `get_imported_files_and_dataframes_from_analysis_name`

It seems like we need the id so we can match the step_id in the function `execute_update_existing_imports_update` at line 42: 

`original_step_index = [index for index, step in enumerate(new_steps) if step.step_id == step_id][0]
`

Reply:
Yeah. I added a comment in the the `get_imported_files_and_dataframes_from_current_steps` and `get_imported_files_and_dataframes_from_analysis_name` functions. See them.

Note that when we're updating import pre replay, we use the replay_analysis api call. When we're updating imports post replay, we use the update_imports call (where we use the step ID).